File: String/Is Subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        if(s.empty()){
            return true;
        }
        if(t.empty()){
            return false;
        }
        int sPtr = 0;
        int tPtr = 0;
        while(sPtr < s.length()){
            if(tPtr > t.length()-1){
                return false;
            }
            if(t[tPtr] == s[sPtr]){
                sPtr++;
            }
            tPtr++;
        }
        return true;
    }
};

File: String/Is Subsequence.rs
impl Solution {
    pub fn is_subsequence(s String, t String) - bool {
        let s_ch Vecchar = s.chars().collect();
        let t_ch Vecchar = t.chars().collect();
        let mut p1 = 0;
        let mut p2 = 0;

        while p1  s_ch.len() && p2  t_ch.len() {
            if s_ch[p1] == t_ch[p2] {
                p1 += 1;
            }
            p2 += 1;
        }

        p1 == s_ch.len()
    }
}